### PR TITLE
feat: (Superadmin)Retrieves emails in waitlist

### PR DESCRIPTION
**Pull request by ericBlack1**
## Description
The PR implements the functionality for getting all emails that were used to signup for the waitlist

## Related Issue (Link to issue ticket)
This project only accepts pull requests related to open issues 
If suggesting a new feature or change, please discuss it in an issue first
If fixing a bug, there should be an issue describing it with steps to reproduce 
Please link to the issue here: [#235](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/235)

## Motivation and Context
This feature provides a comprehensive view of all users who have signed up, enabling superadmins to monitor and manage the waitlist effectively. It will also ensure that only users with superadmin privileges can access this endpoint, maintaining data security and restricting access to authorized personnel.

## How Has This Been Tested?
The functionality has been tested using pytest with the following test cases:

- [x] test_get_all_waitlist_emails_success: Verifies successful waitlist retrieval
- [ ] test_get_all_waitlist_emails_non_superadmin: Verifies for unauthorized access 

## Screenshots (if appropriate - Postman, etc):
Screenshot from 2024-07-29 03-01-59

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/utils/dependencies.py
Comment (joboy-dev) [inline api/utils/dependencies.py:1]: Can you please not use this file to get super admin. There is a function to do that in user service

File: api/v1/routes/waitlist.py [hunk 1]
```diff
@@ -95,4 +96,30 @@ def admin_add_user_to_waitlist(
...
 
-    return JsonResponseDict(**resp)
\ No newline at end of file
+    return JsonResponseDict(**resp)
+
+@waitlist.get("/users", response_model=success_response, status_code=200)
+async def get_all_waitlist_emails(
+    request: Request,
+    db: Session = Depends(get_db),
+    admin=Depends(get_super_admin)
```
Comment (joboy-dev) [inline api/v1/routes/waitlist.py:105]: The get_current_super_admin function in user service should be used here

File: api/v1/routes/waitlist.py [hunk 2]
```diff
@@ -95,4 +96,30 @@ def admin_add_user_to_waitlist(
...
+):
+    try:
```
Comment (joboy-dev) [inline api/v1/routes/waitlist.py:107]: No need for the try and except block. The app exception handler is handling it

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/utils/dependencies.py
Comment (ericBlack1) [reply api/utils/dependencies.py:1]: Okay

File: api/v1/routes/waitlist.py (hunk 2 above)
Comment (ericBlack1) [reply api/v1/routes/waitlist.py]: I've resolved this

File: api/utils/dependencies.py [hunk 3]
```diff
@@ -52,8 +52,12 @@ def get_super_admin(db: Session = Depends(get_db), token: str = Depends(oauth2_s
     if not user.is_super_admin:
         logger.error("User is not a super admin")
         raise HTTPException(
-            status_code=status.HTTP_403_FORBIDDEN,
-            detail="You do not have permission to access this resource",
+            status_code=401,
+            detail={
```
Comment (joboy-dev) [inline api/utils/dependencies.py:56]: Replace this dictionary with what was there before. The exception handler takes care of everything

**Review by joboy-dev — COMMENTED** [review]
Make changes

**Pull request closed without merging**